Src/Fare/RegExp.cs

```diff
@@ -837,10 +837,16 @@ private RegExp ParseSimpleExp()
                 return MakeAnyPrintableASCIIChar();
             }
 
-            if (this.Check(RegExpSyntaxOptions.Empty) && this.Match('#'))
-            {
-                return RegExp.MakeEmpty();
-            }
+            /* Issue 32, https://github.com/moodmosaic/Fare/issues/32
+            *   The intent of the original code is a little unclear.  The comment for the Empty value in the 
+            *   enum is 'Enables empty language.'  Using '#' as token seems non-standard, and caused
+            *   unhandled exception in some cases.  The best option at this point is to remove handling of the
+            *   Empty option until a proper implementation is proposed.
+             */
+            // if (this.Check(RegExpSyntaxOptions.Empty) && this.Match('#'))
+            // {
+            //     return RegExp.MakeEmpty();
+            // }
```

Comment:
FWIW, here's how this looks like in the [Java version](https://github.com/cs-au-dk/dk.brics.automaton/blob/328cf493ec2537af9d2bbce0eb4b4ef118b66547/src/dk/brics/automaton/RegExp.java#L824-L825). Perhaps the same bug exists also there. Happy to accept your fix if it doesn't break anything else.

Reply:
Maybe I should have looked there first! :-p